### PR TITLE
[5.3] Hide empty paginators

### LINKED ISSUE #15124: [5.3] Empty paginator still renders
An empty paginator still renders just the disabled prev/next links or showing only 1 page, seems a bit strange considering in Laravel 5.2 it output nothing.

Is it suggested we publish and modify the view or wrap all our `->render()` calls in an if statement? There's nothing about this mentioned on the upgrade guide so might catch quite a few people out.

image

It also renders when there is only 1 page, which is different from 5.2:

image

### PR DESCRIPTION
**Pull request by garygreen**
Fixes #15124

### CODE REVIEW

**Comment by taylorotwell** [comment]
What is `$elements` and where does that variable come from?

**Comment by taylorotwell** [comment]
"Elements" is only passed by the length aware paginator.

File: src/Illuminate/Pagination/resources/views/bootstrap-4.blade.php [hunk 1]
```diff
@@ -1,34 +1,36 @@
...
-    @else
-        <li class="page-item"><a class="page-link" href="{{ $paginator->previousPageUrl() }}" rel="prev">&laquo;</a></li>
-    @endif
-
-    <!-- Pagination Elements -->
-    @foreach ($elements as $element)
-        <!-- "Three Dots" Separator -->
-        @if (is_string($element))
-            <li class="page-item disabled"><span class="page-link">{{ $element }}</span></li>
+@if (count($elements) > 1)
```
Comment (GrahamCampbell) [inline src/Illuminate/Pagination/resources/views/bootstrap-4.blade.php]: You just just ask the paginator how large it is.

Comment (GrahamCampbell) [inline src/Illuminate/Pagination/resources/views/bootstrap-4.blade.php]: count($paginator)

**Comment by garygreen** [comment]
Not sure why I made that more complicated than it needs to be. I've updated PR to just check for `->count()` on the paginator @taylorotwell @GrahamCampbell :smile:

**Comment by nhowell** [comment]
It still renders when there is only one page but there is more than one item to display, which is different from 5.2.

**Comment by GrahamCampbell** [comment]
Meh, that's ok I think. Kinda odd to hide in that case.

**Comment by garygreen** [comment]
Er so that's why I had the code as I did... I wasn't going mad! I think it should hide it imo, if there's only one page then no point displaying pagination.

**Comment by nhowell** [comment]
@GrahamCampbell I don't think it's ok to not render when there's only one item on the page, while still rendering when there are two items on the page. That's just plain inconsistent.

**Comment by nhowell** [comment]
I would prefer that it doesn't render the pagination links at all unless there is more than one page. Plus, it would make it consistent with 5.2.

But if that's too much trouble, we should at least render the pagination links when there is only one item on the page as well:
```
// Change this:
@if ($paginator->count() > 1)

// To this:
@if (! $paginator->isEmpty())
```

**Comment by GrahamCampbell** [comment]
Oh right I see what you mean. This code should be > 0 not > 1. Please PR. :)

**Comment by nhowell** [comment]
Oh, and I just realized there is another issue with the current code.

Say you used have 26 items and you ```$items->paginate(25)```. The first page has 25 items, while the second page will only have one item and therefore the pagination links will be hidden on the second page.

**Comment by garygreen** [comment]
I think the code should be what I had it as before....

For the non simple paginators it should just be:

```@if (count($elements))```

Then for the simple paginators it should be...

```@if (!$paginator->onFirstPage() || $paginator->hasMorePages())```

**Comment by nhowell** [comment]
@garygreen Oh, I like that.

And in fact, couldn't we use ```@if (!$paginator->onFirstPage() || $paginator->hasMorePages())``` for both types of paginators? I think that would actually retain the same behavior of 5.2 by not rendering at all unless there is more than one page.

**Comment by taylorotwell** [comment]
Reverted. Please send a correct PR.

**Pull request merged**